cypress-io/cypress
# fix: do not duplicate warnings when saving `cypress.config`

- Closes https://github.com/cypress-io/cypress/issues/23341

### User facing changelog


Fix bug where warnings are  duplicated if you save `cypress.config` while a warning is showing.

Eg: https://github.com/cypress-io/cypress/issues/23216#issuecomment-1213493767
Eg: https://github.com/cypress-io/cypress/pull/23034#issuecomment-1209643883

### Additional details


We should store warnings by the type which is unique, not some arbitrary id we generate. We don't want to two show two of the same warnings; we should only show one, the latest one.

Before (observe test on right; it changes the `baseUrl`, and we duplicate a warning):

![image](https://user-images.githubusercontent.com/19196536/184597456-468fa1d5-3596-4bc2-81ae-54795d795982.png)

After (observe test passes - asserts length of warning is 1):

![image](https://user-images.githubusercontent.com/19196536/184597678-6b74270d-0b3e-44a7-bbcc-d7fa7c09de7b.png)


### Steps to test


1. Get this repo
2. `yarn`
3. open some project like `driver`
4. go e2e and go to browser screen
5. change `baseUrl` in `cypress.config.js` to something like `localhost:40404` that does not exist
6. change it to some other ports - it should update the warning with the correct port, and not duplicate the warning
7. change back to `localhost:3500`
8. warning is gone now

### How has the user experience changed?


See above.

### PR Tasks
